ElectronClient/gui/MainScreen.jsx

```diff
@@ -100,14 +100,10 @@ class MainScreenComponent extends React.Component {
 			const folderId = Setting.value('activeFolderId');
 			if (!folderId) return;
 			if (!template) {
-				const templateType = isTodo ? 'defaultTodo' : 'defaultNote';
-				const settingsLabel = Setting.value(templateType);
-				if (settingsLabel != 0) {
-					const matchLabel = this.props.templates.find(i => i.label == settingsLabel);
-					if (!matchLabel) {
-						Setting.setValue(templateType, '0');
-					} else {
-						template = matchLabel.value;
+				const templateFileName = Setting.value(isTodo ? 'editor.defaultTodoTemplate' : 'editor.defaultNoteTemplate');
+				if (templateFileName) {
+					if (typeof(this.props.templates.find(i => i.label == templateFileName)) != 'undefined') {
+						template = this.props.templates.find(i => i.label == templateFileName).value;
```

Comment:
I got rid of Setting.setValue(...). 

After deleting a template and then refreshing templates the Setting.value('editor.default****Template') still returns the name of the deleted file, however the dropdown in settings will default back to 'None'. I made an if (typeof()) test to stop errors getting thrown on new note when a default template has been deleted. 
This also means that if you re-add the deleted template it will be selected as default again, until you actively change to something else in the settings panel.